pkg/authentication/scram/scram.go

```diff
@@ -336,3 +337,8 @@ func convertMongoDBUserToAutomationConfigUser(secretGetUpdateCreateDeleter secre
 	acUser.ScramSha256Creds = &sha256Creds
 	return acUser, nil
 }
+
+// GetConnectionStringSecretName returns the name of the secret where the operator stores the connection string for current user
+func (u User) GetConnectionStringSecretName() string {
+	return fmt.Sprintf("mdbc-%s-%s", u.Database, u.Username)
+}
```

Comment:
I have a question here: could we have a case here where an user has two MongoDBCommunity resources: mdbcA and mdbcB, both with username `foo` on database `foo`, but different passwords, would this generate conflicting secrets?

Reply:
I included the resource name and also implemented spec validation to avoid name collision